dessia_common/schemas/core.py

```diff
@@ -442,8 +480,9 @@ def __init__(self, annotation: Type[T], attribute: str, definition_default: T =
     def to_dict(self, title: str = "", editable: bool = False, description: str = ""):
         """ Write Optional as a Dict. """
         default_value = self.schema.default_value()
-        chunk = self.schema.to_dict(title=title, editable=editable, description=description)
-        chunk["default_value"] = default_value
+
+        base_chunk = self.schema.to_dict(title=title, editable=editable, description=description)
+        chunk = {**base_chunk, "default_value": default_value}
```

Comment:
Elegant !

Reply:
I found this syntax while trying to find a way to prevent pycharm from complaining about return types not validating type hints. I wanted to define that the return type should be an extension of the base chunk, for ex :+1: 

```py
base_chunk = {a: 0, b: 1}
chunk = {**base_chunk, c: 2}
```

But pycharm or mypy doesn't seem to understand that unpacking a dict into another can set the right keys in the result dict which would then validate properly the schema. That's where i should have written : 

`chunk = cast(ExtendedType, {**base_chunk, c: 2})`

Which is the equivalent to typescript's "as", but i am reluctant to do so.

I might reuse this syntax in other schemas